docs/blog/things-to-know-when-developing-with-lit-element/index.md

```diff
@@ -0,0 +1,233 @@
+
+## So you like Typescript?
+
+Typescript loves a good `"Property is not definitely assigned in the constructor"` warning, and if you like Typescript enough to use it a lot, you'll likely run into it at some point. It's telling you this, because in contrast to what you've been told Typescript is NOT smarter than you and it can't tell if something _should_ always be available, only if it _might_ not be available, sometime. You can set an initial value to a property, and it'll never yell:
+
+```js
+class MyThing extends LitElement {
+  myProperty = 'string'; // always available, always a string
+}
+```
+
+If you're looking for that to _have_ to be initialized to use your element, so _you_ KNOW it's going to always have a value, but you want the consumer to initialize it, then you can tell Typescript that being the `!` operator once and be done with it:
+
+```js
+class MyThing extends LitElement {
+  myProperty!: String; // no initial value, but a string is required from the consumer
+}
+```
+
+If you still want to be defensive, you an add some helper code in your lifecycle to support a consumer leveraging your custom element correctly:
+
+```js
+class MyThing extends LitElement {
+  shouldUpdate(changedProperties: PropertyValues<this>): boolean {
+    // great location to make sure it's NEVER undefined;
+    const canUpdate = this.myProperty ?? false;
+    if (!canUpdate) console.warn('`myProperty` is unset');
+    return canUpdate && super.shouldUpdate(changedProperties);
+  }
+  firstUpdated(changedProperties: PropertyValues<this>): void {
+    // for one time availability confirmation
+    super.firstUpdated(changedProperties);
+    if (!this.myProperty) console.warn('`myProperty` is unset');
+  }
+}
+```
+
+### Required properties
+
+In this way, you can also manage required properties/attributes. No, it doesn't fall within the available management of a tool-based contract with the consumers of your component, however, a tool-based contract is not strictly enforceable. You as a component author can tell Typescript or a linter to error on certain things, but you consumer can tell them not to just as easily. Deciding to _only_ leverage tooling for this sort of capability might mean less work for you, but it doesn't guarantee better outcomes for your consumers. Any component author will need to decide the risks they are willing to foist onto their consumers when publishing a component, and this is yet another item to manage on that list.
```

Comment:
"but you consumer can tell them not to" -> "...the consumer..." (or "your consumer"?)